sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/models/ChatRequestAssistantMessage.java

```diff
@@ -58,6 +59,38 @@ public BinaryData getContent() {
         return this.content;
     }
 
+    /**
+     * Get the content property: The contents of the user message, with available input types varying by selected model.
+     * If the result of this method is `null`, it means that the content could be a String or null altogether.
+     *
+     * @return the content value if defined as a list
+     */
+    public List<ChatMessageContentItem> getListContent() {
```

Comment:
We have List, Array, String to represents the content, should we hide the method returns `BinaryData` as it is too general. For example, when user calls `getStringContent()` which return a String, what about a subsequence method call to `getContent()`, returns null value at the moment but String can also convert to BinaryData. We can hide this BinaryData(union of String|List|Array) representation.

Reply:
Hidding that would be an API breaking change and at this point in time, I think people might be relying on this method being public. I would also prefer it removed, given that we have typed alternatives, but yeah, a bit afraid of breaking people with this change.